cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -731,8 +734,20 @@ if(ARROW_WITH_BROTLI)
   include_directories(SYSTEM ${BROTLI_INCLUDE_DIR})
 endif()
 
-if(PARQUET_BUILD_ENCRYPTION OR ARROW_WITH_GRPC)
+set(ARROW_USE_OPENSSL OFF)
+if(ARROW_REQUIRE_ENCRYPTION)
+  # This must work
   find_package(OpenSSL REQUIRED)
+  set(ARROW_USE_OPENSSL ON)
+elseif(ARROW_PARQUET)
+  # Enable Parquet encryption if OpenSSL is there, but don't fail if it's not
+  find_package(OpenSSL QUIET)
+  CMAKE_DEPENDENT_OPTION(ARROW_USE_OPENSSL "Build with OpenSSL support" ON
```

Comment:
I don't think this is the appropriate function. The [doc](https://cmake.org/cmake/help/v3.11/module/CMakeDependentOption.html) says: """This macro presents an option to the user only if a set of other conditions are true."""
But we are always exposing `ARROW_USE_OPENSSL` to the user. I think you simply want to set `ARROW_USE_OPENSSL` conditionally instead.